slice/internal/webhooks/jobset_webhook_test.go

```diff
@@ -21,108 +21,242 @@ import (
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/ptr"
-	ctrl "sigs.k8s.io/controller-runtime"
 	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
-	kueueconstants "sigs.k8s.io/kueue/pkg/controller/constants"
+	kueuealpha "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
 
-	utiltesting "tpu-slice-controller/internal/util/testing"
+	testingjobjobset "tpu-slice-controller/internal/util/testingjobs/jobset"
+	"tpu-slice-controller/test/utils"
 )
 
 func TestDefault(t *testing.T) {
-	testCases := []struct {
-		name    string
-		jobSet  *jobset.JobSet
-		want    *jobset.JobSet
-		wantErr error
+	const (
+		baseJobSetName = "jobset"
+	)
+
+	testCases := map[string]struct {
+		jobSet     *jobset.JobSet
+		wantJobSet *jobset.JobSet
+		wantErr    error
 	}{
-		{
-			name: "TestDefault_No_Local_Queue_label",
-			jobSet: &jobset.JobSet{
-				Spec: jobset.JobSetSpec{
-					ManagedBy: ptr.To(jobset.JobSetControllerName),
-				},
-				ObjectMeta: ctrl.ObjectMeta{
-					Namespace: metav1.NamespaceDefault,
-				},
-			},
-			want: &jobset.JobSet{
-				Spec: jobset.JobSetSpec{
-					ManagedBy: ptr.To(jobset.JobSetControllerName),
-				},
-				ObjectMeta: ctrl.ObjectMeta{
-					Namespace: metav1.NamespaceDefault,
-				},
-			},
+		"no queue label": {
+			jobSet: testingjobjobset.MakeJobSet(baseJobSetName, utils.DefaultNamespace).
+				ReplicatedJobs(testingjobjobset.ReplicatedJobRequirements{
+					Name:        "rj1",
+					Parallelism: 12,
+					PodAnnotations: map[string]string{
+						TPUTopologyAnnotation: "4x4x12",
+						TPUBlockAnnotation:    "cloud.google.com/topology-rack",
+						TPUSubBlockAnnotation: "cloud.google.com/topology-host",
```

Comment:
for the test cases, please use literals instead of variables (here and elsewhere). This makes it easier to see that the test is doing the right thing, without having to scroll, reference the variable

Reply:
also, why are these annotations on the pods? I would only expect TPUTopologyAnnotation and in NodeSelector, TPUAcceleratorLabel

Reply:
> also, why are these annotations on the pods? I would only expect TPUTopologyAnnotation and in NodeSelector, TPUAcceleratorLabel

In result i need to have 3 labels:

```
kueuealpha.PodSetRequiredTopologyAnnotation:      "cloud.google.com/topology-rack",
kueuealpha.PodSetSliceRequiredTopologyAnnotation: "cloud.google.com/topology-host",
kueuealpha.PodSetSliceSizeAnnotation:             "4",
```

IIUC we can get it from:

```
cloud.google.com/gke-tpu-block:    "cloud.google.com/topology-rack",
cloud.google.com/gke-tpu-subblock: "cloud.google.com/topology-host",
```

Reply:
We want something like this:
```
kueue.x-k8s.io/podset-required-topology:  cloud.google.com/gke-tpu-block
kueue.x-k8s.io/podset-slice-required-topology: cloud.google.com/gke-tpu-subblock
kueue.x-k8s.io/podset-slice-size: 16
```

Reply:
Yes, exactly, that what we have.

Reply:
Done